File: DataSets/DataSet-New/LeetCodeTasks - Develop/LeetCodeTasks/3132_find-the-integer-added-to-array-ii/solutions/python3/solution.py
from typing import List
from collections import Counter

class Solution:
    def minimumAddedInteger(self, nums1: List[int], nums2: List[int]) -> int:
        nums2_count = Counter(nums2)
        n = len(nums1)
        min_x = float('inf')
        
        # Try every possible pair to remove from nums1
        for i in range(n):
            for j in range(i + 1, n):
                # Calculate the transformed nums1 without the two selected elements
                transformed = nums1[:i] + nums1[i+1:j] + nums1[j+1:]
                # Attempt to calculate x based on each element matching to nums2
                for k in transformed:
                    for m in nums2:
                        x = m - k
                        # Calculate the new list after applying x
                        new_nums1 = [item + x for item in transformed]
                        if Counter(new_nums1) == nums2_count:
                            min_x = min(min_x, x)
        return min_x

File: DataSets/DataSet-New/LeetCodeTasks - Develop/LeetCodeTasks/3132_find-the-integer-added-to-array-ii/solutions/java/solution.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int minimumAddedInteger(int[] nums1, int[] nums2) {
        Map<Integer, Integer> countMap = new HashMap<>();
        for (int num : nums2) {
            countMap.put(num, countMap.getOrDefault(num, 0) + 1);
        }

        int n = nums1.length;
        int minimumX = Integer.MAX_VALUE;
        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                Map<Integer, Integer> currentCount = new HashMap<>(countMap);
                int candidateX = nums2[0] - nums1[i];
                boolean valid = true;
                
                for (int k = 0; k < n; k++) {
                    if (k != i && k != j) {
                        int transformed = nums1[k] + candidateX;
                        if (currentCount.getOrDefault(transformed, 0) > 0) {
                            currentCount.put(transformed, currentCount.get(transformed) - 1);
                        } else {
                            valid = false;
                            break;
                        }
                    }
                }

                if (valid) {
                    minimumX = Math.min(minimumX, candidateX);
                }
            }
        }

        return minimumX;
    }
}
